joblib/memory.py

```diff
@@ -10,33 +10,29 @@
 
 
 from __future__ import with_statement
+
+import functools
+import inspect
+import io
+import json
```

Comment:
I don't think the pickle import is actually used in memory.py so you can remove it. Could you please leave the imports in the order they were previously?